docs/scripts/gen_reference.py

```diff
@@ -0,0 +1,217 @@
+    )
+
+    spec = schema['properties']['spec']
+    spec_properties = spec['properties']
+    for field_name, field in spec_properties.items():
+        required  = field_name in spec['required'] if 'required' in spec else False
+        res_spec.append(convert_field(field_name, field, required))
+
+    status = schema['properties']['status']
+    status_properties = status['properties']
+    for field_name, field in status_properties.items():
+        required  = field_name in status['required'] if 'required' in status else False
+        res_status.append(convert_field(field_name, field, required))
+
+    return Resource(
+        name=crd['spec']['names']['kind'],
+        service=service, #TODO(RedbackThomson): Load from file path?
+        group=crd['spec']['group'],
+        apiVersion=ver['name'],
+        description=spec['description'],
+        scope=crd['spec']['scope'],
+        names=names,
+        spec=res_spec,
+        status=res_status
+    )
+
+def write_service_pages(service: str, service_path: Path, output_path: Path) -> List[ResourceOverview]:
+    resources = []
+
+    for crd_path in Path(service_path).rglob('*.yaml'):
+        crd = load_crd_yaml(crd_path)
+
+        resource = convert_crd_to_resource(crd, service)
+        resources.append(ResourceOverview(name=resource.name,
+            apiVersion=resource.apiVersion))
+
+        resource_path = output_path / resource.output_path()
+        resource_path.parent.mkdir(parents=True, exist_ok=True)
+
+        with open(resource_path, "w") as out:
+            page_meta = ResourcePageMeta(resource=resource)
+            # TODO(RedbackThomson): Clean up templating
+            print("---", file=out)
+            yaml.dump(asdict(page_meta), out)
+            print("---", file=out)
+            print('{% include "reference.md" %}', file=out)
+    
+    return resources
+
+def write_overview_page(services: List[ServiceOverview], output_path: Path):
+    overview_path = output_path / "overview.md"
+
+    with open(overview_path, "w") as out:
+        page_meta = OverviewPageMeta(services=services)
+        # TODO(RedbackThomson): Clean up templating
+        print("---", file=out)
+        yaml.dump(asdict(page_meta), out)
+        print("---", file=out)
+        print('{% include "reference_overview.md" %}', file=out)
+
+def main(gopath: Path, go_src_parent: Path, service_bases_path: Path, output_path: Path):
+    overviews = []
+
+    src_parent = gopath / go_src_parent
+
+    for controller_repo in src_parent.glob("*-controller"):
+        if not controller_repo.is_dir() or \
+            controller_repo.stem.startswith("template"):
+            continue
+
+        service_bases_path = controller_repo / bases_path
+
+        if not service_bases_path.exists():
+            raise ValueError(f"Service base path {service_bases_path} does not exist")
+
+        # Get service name from repository
+        # TODO(RedbackThomson): Find an elegant way to get name-cased 
```

Comment:
Why? I actually prefer the lowercased alias because that is what will appear in both the name of the controller image/chart tags as well as the CRD's APIGroup...

Reply:
I think we'll need both. One for prettiness and one for usability. But it's always easier to go from proper casing to lowercasing?

Reply:
well, either way it's not urgent :)